shared/modules/treestore_module.go

```diff
@@ -17,6 +17,7 @@ type TreeStoreFactory = FactoryWithOptions[TreeStoreModule, TreeStoreOption]
 // merkle trees that compose the state hash of pocket.
 type TreeStoreModule interface {
 	Submodule
+	TreeStoreFactory
```

Comment:
`TreeStoreFactory` shouldn't need to be exported

Reply:
👍